urllib3/connection.py

```diff
@@ -470,7 +466,7 @@ class HTTPConnection(object):
     response_class = OldHTTPResponse
 
     def __init__(self, host, port, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
-                 source_address=None, socket_options=_UNKNOWN):
+                 source_address=None, socket_options=default_socket_options):
```

Comment:
So I went with `None` originally for this, and the problem is that `None` has different semantics than `_UNKNOWN`.

Passing `None` was previously equivalent to `self.socket_options = None`, but 
```python
self.socket_options = (
             socket_options if socket_options is not None		
             else self.default_socket_options		
)
```
means that that it falls to the else statement and `self.socket_options = self.default_socket_options`. 

I think we'll either need another sentinel object for this specific scenario, or change how we signify "No Socket Options".